examples/oapp-solana/programs/my_oapp/src/msg_codec.rs

```diff
@@ -0,0 +1,35 @@
+
+pub const VANILLA_TYPE: u8 = 1;
+
+// Just like OFT, we don't need an explicit MSG_TYPE param
+// Instead, we'll check whether there's data after the string ends
+pub const LENGTH_OFFSET: usize = 0;
+pub const STRING_OFFSET: usize = 32;
+
+fn decode_string_len(buf: &[u8]) -> usize {
+    let mut string_len_bytes = [0u8;32];
+    string_len_bytes.copy_from_slice(&buf[LENGTH_OFFSET..LENGTH_OFFSET+32]);
+    u32::from_be_bytes(string_len_bytes[28..32].try_into().unwrap()) as usize
+}
```

Comment:
The `decode_string_len` function assumes the input buffer is at least 32 bytes long to safely read the length field. However, there's no validation of the buffer length before accessing it.
Suggested Fix:
```Rust
fn decode_string_len(buf: &[u8]) -> Result<usize, Error> {
    if buf.len() < 32 {
        return Err(MsgCodecError::InvalidLength);
    }
    let mut string_len_bytes = [0u8;32];
    string_len_bytes.copy_from_slice(&buf[LENGTH_OFFSET..LENGTH_OFFSET+32]);
    Ok(u32::from_be_bytes(string_len_bytes[28..32].try_into().unwrap()) as usize)
}
```